src/clj/game/cards/upgrades.clj

```diff
@@ -1838,10 +1838,14 @@
                  (corp? (:card target))
                  (installed? (:card target))))]
     {:on-trash {:async true
-                :once-per-instance false
                 :interactive (req true)
-                :msg "gain 2 [Credits]"
-                :effect (effect (gain-credits eid 2))}
+                :effect (req (doseq [t targets]
```

Comment:
Can't call `doseq` on async functions. You need to either use recursion to gain the credits one at a time or do something like `(gain-credits state side eid (filter #(and ...) (map :card targets)))`.

Reply:
I wonder if this ability even needs to be async anymore (or in the first place)? I might have just inherited it from its old implementation? Maybe `:interactive` is also unnecessary?

If still required, do you have an example of where we do this with recursion? I don't know if the `args` route will work with gain-credits since I want to provide a message for each credit gain as well.